### PR TITLE
BHoM: Make oM properties virtual

### LINKED ISSUE #777: Add virtual keyword to oM properties
In line with the discussion on https://github.com/BHoM/BHoM/issues/733 and subsequent changes to BHoM in https://github.com/BHoM/BHoM/issues/734 it is desirable for all objects to now contain the `virtual` modifier to provide greater flexibility for developers and users wishing to create BHoM objects in their workflows.

This issue is to instruct @BHoMBot to carry out the task of adding the `virtual` modifier to any object in the oM of this repository where it does not already exist.

### PR DESCRIPTION
**Pull request by BHoMBot**
Closes #777

### CODE REVIEW

File: Structure_oM/Results/Global Results/ModalDynamics.cs [hunk 1]
```diff
@@ -33,7 +33,7 @@ public class ModalDynamics : StructuralGlobalResult
         /***************************************************/
 
         
-        public int ModeNumber { get; set; } = 0;
+        public virtual int ModeNumber { get; set; } = 0;
 
         [Frequency]
         public double Frequency { get; set; } = 0.0;
```
Comment (al-fisher) [inline Structure_oM/Results/Global Results/ModalDynamics.cs:39]: Missing Virtual noticed of some Properties in this file.

File: Structure_oM/SurfaceProperties/ConstantThickness.cs [hunk 2]
```diff
@@ -36,13 +36,13 @@ public class ConstantThickness : BHoMObject, ISurfaceProperty
...
         [Description("Defines the distance the element should be extruded along its normal. By default the element will be extruded half the thickness 'upwards' and half the thickness 'downwards' meaning the element base geoemtry will be in the centre of the extrusion.")]
-        public double Thickness { get; set; }
+        public virtual double Thickness { get; set; }
 
         [Description("Homogenous structural material throughout the full thickness of the element.")]
-        public IMaterialFragment Material { get; set; }
+        public virtual IMaterialFragment Material { get; set; }
 
         [Description("Defines what type of element this property will be used. Used by some analysis packages.")]
-        public PanelType PanelType { get; set; } = PanelType.Slab;   //TODO: Required to get Etabs working. To be moved to physical objects
```
Comment (al-fisher) [inline Structure_oM/SurfaceProperties/ConstantThickness.cs:45]: Question to @IsakNaslundBh - the inline comment has been purged as part of this process (as result of using the sytaxtree representation) - is this a problem for you?

File: Structure_oM/SectionProperties/Reinforcement/Reinforcement.cs [hunk 3]
```diff
@@ -31,15 +31,15 @@ public abstract class Reinforcement : BHoMObject
...
-        public double Diameter { get; set; }
+        public virtual double Diameter { get; set; }
 
-        public int BarCount { get; set; }
+        public virtual int BarCount { get; set; }
 
-        public Material Material { get; set; }
+        public virtual Material Material { get; set; }
 
-        public double StartLocation { get; set; } = 0;  // location of the beginning of the reinforcement as a ratio of the bar length
```
Comment (al-fisher) [inline Structure_oM/SectionProperties/Reinforcement/Reinforcement.cs:40]: @IsakNaslundBh see other comment regarding removal of inline comments - non-compliant obviously. Are you happy to let these be deleted?

File: Structure_oM/SurfaceProperties/ConstantThickness.cs (hunk 2 above)
Comment (IsakNaslundBh) [reply Structure_oM/SurfaceProperties/ConstantThickness.cs:45]: This case is fine. Think we have to live with the panel type for the time being, so fine with this comment being removed.

File: Structure_oM/SectionProperties/Reinforcement/Reinforcement.cs (hunk 3 above)
Comment (IsakNaslundBh) [reply Structure_oM/SectionProperties/Reinforcement/Reinforcement.cs:40]: Same here, these classes has not been documented yet, as they are to be completely rewritten from the ground up. Any comments in here are more than fine to be deleted.

**Pull request merged**